addons/mrp_subcontracting/models/stock_picking.py

```diff
@@ -41,47 +41,59 @@ def _compute_display_action_record_components(self):
     # -------------------------------------------------------------------------
     def _action_done(self):
         res = super(StockPicking, self)._action_done()
+
+        for move in self.move_lines.filtered(lambda move: move.is_subcontract):
+            # Auto set qty_producing/lot_producing_id of MO if there isn't tracked component
+            # If there is tracked component, the flow use subcontracting_record_component instead
+            if move._has_tracked_subcontract_components():
+                continue
+            production = move.move_orig_ids.production_id.filtered(lambda p: p.state not in ('done', 'cancel'))[-1:]
+            if not production:
+                continue
+            # Manage additional quantities
+            quantity_done_move = move.product_uom._compute_quantity(move.quantity_done, production.product_uom_id)
+            if float_compare(production.product_qty, quantity_done_move, precision_rounding=production.product_uom_id.rounding) == -1:
+                change_qty = self.env['change.production.qty'].create({
+                    'mo_id': production.id,
+                    'product_qty': quantity_done_move
+                })
+                change_qty.with_context(skip_activity=True).change_prod_qty()
+            # Create backorder MO for each move lines
+            for move_line in move.move_line_ids:
+                # The first MO will have the qty_producing == 0.0
+                if not float_is_zero(production.qty_producing, precision_rounding=production.product_uom_id.rounding):
+                    backorder = production._generate_backorder_productions(close_mo=False)
+                    # The move_dest_ids won't be set because the _split filter out done move
+                    backorder.move_finished_ids.filtered(lambda mo: mo.product_id == move.product_id).move_dest_ids = production.move_finished_ids.filtered(lambda mo: mo.product_id == move.product_id).move_dest_ids
+                    production.product_qty = production.qty_producing
+                    production = backorder
+                if move_line.lot_id:
+                    production.lot_producing_id = move_line.lot_id
```

Comment:
Is it intended to set the lot on `production` after doing `production = backorder` ? I've the idea the first mo doesn't get a lot_producing_id

Reply:
```python
            for move_line in move.move_line_ids:
                if move_line.lot_id:
                    production.lot_producing_id = move_line.lot_id
                production.qty_producing = move_line.product_uom_id._compute_quantity(move_line.qty_done, production.product_uom_id)
                production._set_qty_producing()
# Si tu veux tu peux ici tester si tu es en train de traiter la dernière move line 
                backorder = production._generate_backorder_productions(close_mo=False)
                # The move_dest_ids won't be set because the _split filter out done move
                backorder.move_finished_ids.filtered(lambda mo: mo.product_id == move.product_id).move_dest_ids = production.move_finished_ids.filtered(lambda mo: mo.product_id == move.product_id).move_dest_ids
                production.product_qty = production.qty_producing
                production = backorder
                
```